### PR TITLE
If prepare rename timeouts, the rename operation can still be done,

### PR DESCRIPTION
**Pull request by rubenporras**
"newName" will be used as placeholder. Adding a fatal error is not necessary.

### CODE REVIEW

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameProcessor.java [hunk 1]
```diff
@@ -110,7 +111,11 @@ public RefactoringStatus checkInitialConditions(IProgressMonitor pm)
 					final var params = new PrepareRenameParams();
 					params.setTextDocument(identifier);
 					params.setPosition(LSPEclipseUtils.toPosition(offset, document));
-					prepareRenameResult = languageServer.getTextDocumentService().prepareRename(params).get(1000, TimeUnit.MILLISECONDS);
+					try {
+						prepareRenameResult = languageServer.getTextDocumentService().prepareRename(params).get(1000, TimeUnit.MILLISECONDS);
+					} catch (TimeoutException e) {
+						LanguageServerPlugin.logInfo("Could not prepare rename due to timeout after 1 seconds in `textDocument/prepareRename`. 'newName' will be used"); //$NON-NLS-1$
```
Comment (mickaelistria) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameProcessor.java:117]: This should be a warning as it's abnormal behavior.

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameProcessor.java [hunk 2]
```diff
@@ -110,7 +111,11 @@ public RefactoringStatus checkInitialConditions(IProgressMonitor pm)
...
 					params.setTextDocument(identifier);
 					params.setPosition(LSPEclipseUtils.toPosition(offset, document));
-					prepareRenameResult = languageServer.getTextDocumentService().prepareRename(params).get(1000, TimeUnit.MILLISECONDS);
+					try {
+						prepareRenameResult = languageServer.getTextDocumentService().prepareRename(params).get(1000, TimeUnit.MILLISECONDS);
+					} catch (TimeoutException e) {
+						LanguageServerPlugin.logInfo("Could not prepare rename due to timeout after 1 seconds in `textDocument/prepareRename`. 'newName' will be used"); //$NON-NLS-1$
+					}
 					if (prepareRenameResult == null) {
 						status.addFatalError(Messages.rename_invalidated);
```
Comment (mickaelistria) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameProcessor.java:120]: Isn't it still blocking the operation in such case?

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameProcessor.java (hunk 1 above)
Comment (rubenporras) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameProcessor.java]: Okay, I have changed it. It is only that this warnings with an exception attached look a bit scary on the console, but that is another topic.

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameProcessor.java (hunk 2 above)
Comment (rubenporras) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameProcessor.java]: done

**Pull request merged**